rmf_site_format/src/workcell.rs

```diff
@@ -289,6 +344,110 @@ impl Workcell {
         serde_json::ser::to_string_pretty(self)
     }
 
+    pub fn to_urdf(&self) -> Result<urdf_rs::Robot, WorkcellToUrdfError> {
+        let workcell = self.clone();
+
+        let visuals_and_parents: Vec<(urdf_rs::Visual, _)> = workcell
+            .visuals
+            .into_iter()
+            .map(|v| {
+                let visual = v.1.bundle;
+                let visual = urdf_rs::Visual {
+                    name: Some(visual.name),
+                    origin: visual.pose.into(),
+                    geometry: visual.geometry.into(),
+                    material: None,
+                };
+                (visual, v.1.parent)
+            })
+            .collect();
+
+        let collisions_and_parents: Vec<(urdf_rs::Collision, _)> = workcell
+            .collisions
+            .into_iter()
+            .map(|c| {
+                let collision = c.1.bundle;
+                let collision = urdf_rs::Collision {
+                    name: Some(collision.name),
+                    origin: collision.pose.into(),
+                    geometry: collision.geometry.into(),
+                };
+                (collision, c.1.parent)
+            })
+            .collect();
+
+        let links: Result<Vec<_>, WorkcellToUrdfError> = workcell
+            .frames
+            .into_iter()
+            .map(|f| {
+                let frame = f.1.bundle;
+
+                let name = match frame.name {
+                    Some(name) => name.0,
+                    None => format!("frame_{}", f.0),
+                };
+                let visual: Vec<Visual> = visuals_and_parents
+                    .iter()
+                    .filter(|(_, parent)| parent == &f.1.parent)
+                    .map(|(visual, _)| visual.clone())
+                    .collect();
+                let collision: Vec<Collision> = collisions_and_parents
+                    .iter()
+                    .filter(|(_, parent)| parent == &f.1.parent)
+                    .map(|(collision, _)| collision.clone())
+                    .collect();
+
+                let pose: urdf_rs::Pose;
+                if let Anchor::Pose3D(p) = frame.anchor {
+                    pose = p.into();
+                } else {
+                    return Err(WorkcellToUrdfError::InvalidAnchorType(frame.anchor));
+                }
```

Comment:
This can be "simplified" using the return type of the `if let` expression, such as:

```
let pose = if let Anchor::Pose3D(p) = frame.anchor {
  p.into()
} else {
  return Err(WorkcellToUrdfError::InvalidAnchorType(frame.anchor));
};
```

Reply:
I like that much more. Thanks for letting me know!

Reply:
It's amazing that it doesn't need a type here and that it infers it from the usage later in the code. Magic.

Reply:
Done in 77883bd.